Mastodon/Supporting Files/AppDelegate.swift

```diff
@@ -64,13 +64,6 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 }
 
 extension AppDelegate {
-    func application(_ application: UIApplication, supportedInterfaceOrientationsFor window: UIWindow?) -> UIInterfaceOrientationMask {
-        return UIDevice.current.userInterfaceIdiom == .phone ? .portrait : .all
-    }
-}
-
-extension AppDelegate {
-    
```

Comment:
What happens due to this, is that several other screens get rotated as well, like the Onboarding or Safari:

<img widht="300px" src="https://user-images.githubusercontent.com/2580019/209922391-b916c293-8248-4304-9942-ad7765bf3144.png">

<img widht="300px" src="https://user-images.githubusercontent.com/2580019/209922383-2e31e6af-f96b-4818-a485-56cba3d0aeba.png">

Sure, one could just add this to every affected screen again and again: 

```swift
override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
    return UIDevice.current.userInterfaceIdiom == .phone ? .portrait : .all
}
```

I know: Rotation sucks. But I keep thinking: Is there a another way, a more elegant way to achieve rotation for one single screen and not change other existing ones? Like using a protocol? Or maybe a extra component, that determines (and returns orientation), if a screen should be rotated?

Something like [this](https://stackoverflow.com/a/44055120)?

Reply:
Maybe! (Or it could go the other way and have a protocol that implements that method to constrain orientations for us). That said I think at least the Safari change is an improvement. Sometimes websites (especially ones with videos) need to be viewed in landscape.

Reply:
Regarding Safari: Fair point, yes, I'm convinced 👍  It just changed something, I wasn't  sure, if this was intended, as the app itself is in portrait only (on phone). But the broken onboarding is still an issue :)

Reading from the sensor or using an overlay to fake something feels either pretty overkill or pretty hacky (or both). If there's no better way, then that's what it is :D